### ISSUE TITLE
Issue on the Colab

### ISSUE THREAD

**Issue by chisu7917** [eloquentarduino/EloquentTinyML#41]
Hi Simone,

Thanks for your very nice code! Really appreciate it!
I have the same issue with Radhanand Anantha who post a question on your blog.  https://eloquentarduino.github.io/2020/11/tinyml-on-arduino-and-stm32-cnn-convolutional-neural-network-example/#tochow-to-run-a-cnn-on-arduino-and-stm32-boards-with-eloquenttinyml
I also use colab to generate the model and finally the model size is about 5000 instead of 21448. The code is totally the same with yours because I just copied it to verify. 
Here is the link of colab code. Could you please have a look? Thanks!
https://colab.research.google.com/drive/1_KE3OJ5uWD4bD-DITsQ4N6J8N8KDRorT?usp=sharing

**Comment by eloquentarduino** [comment]
Does the model of size 5000 works fine or it is not able to classify digits?

**Comment by chisu7917** [comment]
Hi Simone,

No, it doesn't work. it looks like as below. 

18:40:37.835 -> It took 3 micros to run inference
18:40:37.835 -> Test output is: 8
18:40:37.835 -> Predicted proba are: 0.00,0.00,0.00,0.00,0.00,0.00,0.00,0.00,0.00,0.00
18:40:37.835 -> Predicted class is: 0
18:40:37.835 -> Sanity check: 7

**Comment by eloquentarduino** [comment]
You probably have an outdated version of tinymlgen. Either

 1. upgrade tinymlgen
 2. use `port(model, variable_name='digits_model', pretty_print=True, optimize=False)`

The latest version of tinymlgen makes `optimize=False` the default, prior versions had `optimize=True`.
It is now working fine for me.

**Comment by chisu7917** [comment]
Thanks for your prompt reply, Simone!

Yes, you are right. I have modified the code and got the right results of classification!! I tried both methods your mentioned and got the same result.

I'm still curious why the length of my model is "digits_model_len = 13504;" instead of 21448? Both your methods generated the same length of model of 13504 on my side.

Thanks!

https://colab.research.google.com/drive/1_KE3OJ5uWD4bD-DITsQ4N6J8N8KDRorT?usp=sharing
